inst/components/scss/sidebar.scss

```diff
@@ -315,28 +318,68 @@ $bslib-sidebar-column-sidebar: Min(calc(100% - var(--_padding-icon)), var(--_sid
     }
 
     &[data-collapsible-mobile="true"] {
+      // On mobile, when the sidebar is collapsible, we add an additional row
+      // for the collapse toggle. This avoid overlapping the toggle and the
+      // main content when the sidebar is collapsed. When the sidebar expands,
+      // it takes up the entire layout area.
+      grid-template-rows: calc(var(--_icon-button-size) + var(--_padding)) 1fr;
+      > .collapse-toggle {
+        grid-row: 1 / 2; // top row
+      }
+      > .main {
+        grid-row: 2 / 3; // bottom row
+      }
+      > .sidebar {
+        grid-row: 1 / 3; // whole layout
+      }
       // Sidebar layer has to be lifted up to cover other (nested) sidebars
-      &:not(.sidebar-collapsed), &.transitioning {
-        > .sidebar { z-index: $zindex-offcanvas; }
-        > .collapse-toggle { z-index: $zindex-offcanvas; }
+      &:not(.sidebar-collapsed),
+      &.transitioning {
+        > .sidebar {
+          z-index: $zindex-offcanvas;
+        }
+        > .collapse-toggle {
+          z-index: $zindex-offcanvas;
+        }
       }
 
+      > .collapse-toggle {
+        top: unset;
+        position: relative;
+        align-self: center;
+      }
+        
       // Keep toggle on sidebar side when expanded on mobile
       &:not(.sidebar-right) > .collapse-toggle {
         left: var(--_icon-size);
         right: unset;
+        justify-self: left;
       }
 
       &.sidebar-right > .collapse-toggle {
         right: var(--_icon-size);
         left: unset;
+        justify-self: right;
       }
 
       > .sidebar {
+        backdrop-filter: blur(6px);
+        -webkit-backdrop-filter: blur(6px);
```

Comment:
I'm going to take this out; it doesn't have an effect currently so it's not worth potential performance effects.

```suggestion
```